File: cpp/program4.cpp
// program23.c

#include<iostream>
using namespace std;

class Factors
{
    public:
       int iNo;

    Factors(int X)  
    {
        iNo = X;
    }

    void DisplayFactors()
    {
        int iCnt = 0;

        for(iCnt = 1; iCnt < (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                cout<<iCnt<<"\n";
            }
        }
    }
};

int main()
{
    int iValue = 0;
    
    cout<<"Enter number "<<"\n";
    cin>>iValue;

    Factors fobj(iValue);
    fobj.DisplayFactors();

    return 0;
}

File: java/program4.java

import java.util.*;

class Number
{
    public int iNo;

    public Number(int X)
    {
        iNo = X;
    }

    public void DisplayFactor()
    {
        for(int iCnt = 1; iCnt <= (iNo/2); iCnt++)
        {
            if((iNo % iCnt) == 0)
            {
                System.out.println(iCnt);
            }
        }
    }
}

class program4
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number ");
        int iValue = sobj.nextInt();

        Number obj = new Number(iValue);
        obj.DisplayFactor();
    }
}